dotnet/src/SemanticKernel/SemanticFunctions/PromptTemplate.cs

```diff
@@ -64,16 +63,6 @@ public IList<ParameterView> GetParameters()
             result[p.Name] = new ParameterView(p.Name, p.Description, p.DefaultValue);
         }
 
-        // Parameters from the template
-        foreach (var block in this._templateEngine.ExtractBlocks(this._template))
-        {
-            string? blockName = (block as VarBlock)?.Name;
-            if (!string.IsNullOrEmpty(blockName) && !result.ContainsKey(blockName!))
-            {
-                result.Add(blockName!, new ParameterView(blockName!));
-            }
-        }
-
```

Comment:
Removing this logic can bring some strong consequences.

Shouldn't we have a `GetParameters()` in the `_templateEngine` `IPromptTemplateEngine` interface? 

So this call would be something like:

```
foreach (var p in this._templateEngine.GetParameters())
{
     result[p.Name] = new ParameterView(p.Name, p.Description, p.DefaultValue);
}
```

Reply:
I made that change in an earlier commit and when I reviewed with Shawn the decision was made that we would remove this code if it doesn't cause any regressions. This code is a bit hacky and likely cannot be easily support by other template engines. So in the interest of keeping things simple we are removing it.

Reply:
#3115